src/Omnipay/Common/Message/AbstractResponse.php

```diff
@@ -107,9 +107,24 @@ public function getCode()
         return null;
     }
 
+    /**
+     * Get the transaction reference as generated by the remote payment gateway.
+     *
+     * @return string
+     */
     public function getTransactionReference()
     {
-        return null;
```

Comment:
@kayladnls Any idea why this was originally null?

I understand `getTransactionId()` was never in the original `Response` message because not all responses involve the transaction ID, but a good few do, so it may make sense to introduce it. Don't have any quantitative stats on how many drivers define `getTransactionId()` in their general Response message.